src/platform/utilities/sso/index.js

```diff
@@ -0,0 +1,51 @@
+import * as Sentry from '@sentry/browser';
+
+import environment from 'platform/utilities/environment';
+import localStorage from '../storage/localStorage';
+import { hasSessionSSO } from '../../user/profile/utilities';
+import mockKeepAlive from './mockKeepAliveSSO';
+import liveKeepAlive from './keepAliveSSO';
+
+const keepAlive = environment.isLocalhost() ? mockKeepAlive : liveKeepAlive;
+let ssoSessionLength = 9000; // milliseconds
+
+export async function ssoKeepAliveSession() {
+  const res = await keepAlive();
+
+  try {
+    const hasSSOsession = res.headers.get('session-alive');
```

Comment:
the `session-alive` and `session-timeout` headers are specific to the `keepAlive` implementation.  Did you ever think about just having `keepAlive` function return a promise with the timeout value?

Reply:
right now `keepAlive()` is very minimal, just a fetch call to with the proper config params. I split that out mainly so I could have that vs a mocked call for local development.

my thinking was this is the primary implementation layer for that logic, since it's isolated to that single purpose and hides the keep-alive details from other things calling it (ie, all the timeout logic is contained here).

Reply:
yeah I think splitting it out was a good idea, and if the function was just returing an integer rather than a HTTP response, its also easy to mock.  my thought was that if the IAM team ever changes the implementation or asks us to check the "status" in another way, we'd only have to change the function in keepAliveSSO.js.  I'm good either way, just wanted to run that idea by you.

Reply:
it's a good idea, I'll play around with it while waiting for other reviewers.